pkg/ipamd/ipamd.go

```diff
@@ -452,6 +452,9 @@ func New(rawK8SClient client.Client, cachedK8SClient client.Client) (*IPAMContex
 		go wait.Forever(func() { _ = c.awsClient.RefreshSGIDs(mac) }, 30*time.Second)
 	}
 
+	// setup Host instanceID in event recorder
+	eventrecorder.HostID = c.awsClient.GetInstanceID()
```

Comment:
Can you move this up in line 397 but make sure to also reorder the `awsClient` in 405. You might need to move the event recorder init below 405..

Reply:
Makes sense. I have updated the PR. Please review. Thanks.